dids/resolution.go

```diff
@@ -0,0 +1,75 @@
+package dids
+
+// ResolutionResult represents the result of a DID (Decentralized Identifier)
+// resolution.
+//
+// This class encapsulates the metadata and document information obtained as
+// a result of resolving a DID. It includes the resolution metadata, the DID
+// document (if available), and the document metadata.
+//
+// The `DidResolutionResult` can be initialized with specific metadata and
+// document information, or it can be created with default values if no
+// specific information is provided.
+type ResolutionResult struct {
+	// The metadata associated with the DID resolution process.
+	//
+	// This includes information about the resolution process itself, such as any errors
+	// that occurred. If not provided in the constructor, it defaults to an empty object
+	// as per the spec
+	ResolutionMetadata ResolutionMetadata `json:"didResolutionMetadata,omitempty"`
+	// The resolved DID document, if available.
+	//
+	// This is the document that represents the resolved state of the DID. It may be `null`
+	// if the DID could not be resolved or if the document is not available.
+	Document Document `json:"didDocument"`
+	// The metadata associated with the DID document.
+	//
+	// This includes information about the document such as when it was created and
+	// any other relevant metadata. If not provided in the constructor, it defaults to an
+	// empty `DidDocumentMetadata`.
+	DocumentMetadata DocumentMetadata `json:"didDocumentMetadata,omitempty"`
```

Comment:
It would be idiomatic to make a non-required field into a pointer. It's not as ideal from a correctness standpoint as using an option type, but it's fairly idiomatic.